jmx-metrics/src/integrationTest/java/io/opentelemetry/contrib/jmxmetrics/target_systems/JettyIntegrationTest.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.jmxmetrics.target_systems;
+
+import static org.assertj.core.api.Assertions.entry;
+
+import io.opentelemetry.contrib.jmxmetrics.AbstractIntegrationTest;
+import java.io.IOException;
+import java.time.Duration;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.junit.jupiter.Container;
+
+class JettyIntegrationTest extends AbstractIntegrationTest {
+
+  JettyIntegrationTest() {
+    super(/* configFromStdin= */ false, "target-systems/jetty.properties");
+  }
+
+  @Container
+  GenericContainer<?> jetty =
+      new GenericContainer<>(
+              new ImageFromDockerfile()
+                  .withDockerfileFromBuilder(
+                      builder ->
+                          builder
+                              .from("jetty")
+                              .run(
+                                  "java",
+                                  "-jar",
+                                  "/usr/local/jetty/start.jar",
+                                  "--add-to-startd=jmx,stats,http")
+                              .env(
+                                  "JAVA_OPTIONS",
+                                  "-Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.port=1099")
+                              .build()))
+          .withNetwork(Network.SHARED)
+          .withEnv("LOCAL_JMX", "no")
+          .withNetworkAliases("jetty")
+          .withExposedPorts(1099)
+          .withStartupTimeout(Duration.ofMinutes(2))
+          .waitingFor(Wait.forListeningPort());
+
+  @Test
+  void endToEnd() throws IOException {
+    waitAndAssertMetrics(
+        // Load on the server is required to test session metrics
```

Comment:
You could have a test client container making requests like is done w/ the kafka topic creator and producer in its integration tests.